site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -203,3 +203,138 @@ WHERE id = (SELECT id
+
+```sql
+-- SELECT iasql_install('aws_acm'); -- no need for this because aws_route53 depends on it and it's automatically installed
+SELECT certificate_request('<your-domain>', 'DNS', default_aws_region(), '{}');
+
+-- inspect the new certificate
+SELECT *
+FROM certificate
+WHERE domain_name = '<your-domain>' AND status = 'ISSUED';
+```
+
+The second query confirms that we have a new certificate for our domain, and it's ready to be used (status = `ISSUED`). Now we're ready to launch our load balancer, listen for HTTPS traffic with this new certificate, and forward the incoming traffic to our EC2 instance's 9876 port.
+
+#### Spawn An Elastic Load Balancer
+To set up an elastic load balancer, we need some kinds of resources:
+- `load_balancer` itself
+- A `target_group` pointing to our EC2 instance on port 9876
+- A `listener` on port 443 of the load balancer
+
+<img src={'/img/iasql-on-ec2/load-balancer-parts.png'} style={{maxWidth: 300}} />
+
+You know the drill: we `INSERT` rows in the corresponding tables, and IaSQL will apply our desired changes to the cloud (remember to replace `<your-ip>` with your own public IP address).
+
+First, we create a new security group for our load balancer that should allow traffic on 443 port only from our IP (while allowing outgoing traffic from our load balancer):
+```sql
+INSERT INTO security_group (description, group_name, region)
+VALUES ('allows traffic on 443 port from my ip', 'iasql-lb', default_aws_region());
+
+WITH lb_security_group as (SELECT id FROM security_group WHERE group_name = 'iasql-lb')
+INSERT
+INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id, region)
+VALUES (false, 'tcp', 443, 443, '<your-ip>/32', 'allow incoming traffic on 443 from my ip',
+        (SELECT id FROM lb_security_group), default_aws_region()),
+       (true, '-1', -1, -1, '0.0.0.0/32', 'allow outgoing traffic from load balancer',
+        (SELECT id FROM lb_security_group), default_aws_region());
+```
+
+Then perform creation of the actual load balancer, the target group, and the listener:
+```sql
+INSERT INTO load_balancer (load_balancer_name, scheme, load_balancer_type, ip_address_type, region)
+VALUES ('iasql-lb', 'internet-facing', 'application', 'ipv4', default_aws_region());
+
+INSERT INTO target_group (target_group_name, target_type, protocol, port)
```

Comment:
should not be HTTP? listener is HTTPS but traffic inside VM is just HTTP right?

Reply:
Yes you're right. I'll fix it.